ambry-cloud/src/main/java/com.github.ambry.cloud/CloudBlobStore.java

```diff
@@ -0,0 +1,250 @@
+  private CloudDestination cloudDestination;
+  private boolean started;
+
+  /**
+   * Constructor for CloudBlobStore
+   * @param partitionId partition associated with BlobStore.
+   * @param verProps the cloud store properties.
+   */
+  CloudBlobStore(PartitionId partitionId, VerifiableProperties verProps, CloudDestination cloudDestination) {
+    this.cloudDestination = cloudDestination;
+    this.partitionId = partitionId;
+  }
+
+  @Override
+  public void start() throws StoreException {
+    started = true;
+  }
+
+  @Override
+  public StoreInfo get(List<? extends StoreKey> ids, EnumSet<StoreGetOptions> storeGetOptions) throws StoreException {
+    throw new UnsupportedOperationException("Method not supported");
+  }
+
+  @Override
+  public void put(MessageWriteSet messageSetToWrite) throws StoreException {
+    checkStarted();
+    if (messageSetToWrite.getMessageSetInfo().isEmpty()) {
+      throw new IllegalArgumentException("Message write set cannot be empty");
+    }
+    checkDuplicates(messageSetToWrite);
+
+    // Write the blobs in the message set
+    CloudWriter cloudWriter = new CloudWriter(this, messageSetToWrite.getMessageSetInfo());
+    try {
+      messageSetToWrite.writeTo(cloudWriter);
+    } catch (IOException ex) {
+      throw new StoreException(ex, StoreErrorCodes.IOError);
+    }
+  }
+
+  /**
+   * Upload the blob to the cloud destination.
+   * @param messageInfo the {@link MessageInfo} containing blob metadata.
+   * @param messageBuf the bytes to be uploaded.
+   * @param size the number of bytes to upload.
+   * @throws Exception
+   */
+  private void putBlob(MessageInfo messageInfo, ByteBuffer messageBuf, long size) throws Exception {
+    boolean performUpload = messageInfo.getExpirationTimeInMs() == Utils.Infinite_Time && !messageInfo.isDeleted();
+    // May need to encrypt buffer before upload
+    if (performUpload) {
+      BlobId blobId = (BlobId) messageInfo.getStoreKey();
+      cloudDestination.uploadBlob(blobId, size, new ByteBufferInputStream(messageBuf));
+    }
+  }
+
+  @Override
+  public void delete(MessageWriteSet messageSetToDelete) throws StoreException {
+    checkStarted();
+    checkDuplicates(messageSetToDelete);
+
+    try {
+      for (MessageInfo msgInfo : messageSetToDelete.getMessageSetInfo()) {
+        BlobId blobId = (BlobId) msgInfo.getStoreKey();
+        cloudDestination.deleteBlob(blobId);
+      }
+    } catch (Exception ex) {
+      throw new StoreException(ex, StoreErrorCodes.IOError);
+    }
+  }
+
+  /**
+   * {@inheritDoc}
+   * Currently, the only supported operation is to set the TTL to infinite (i.e. no arbitrary increase or decrease)
+   * @param messageSetToUpdate The list of messages that need to be updated
+   * @throws StoreException if there is a problem persisting the operation in the store.
+   */
+  @Override
+  public void updateTtl(MessageWriteSet messageSetToUpdate) throws StoreException {
+    checkStarted();
```

Comment:
My thinking is that findMissingKeys() will return the blobs corresponding to the UpdateTTL record, because they weren't uploaded to Azure when the Put happened.  So then the source replica will send the blob data over to the VCR.  Will it not work that way?